packages/roosterjs-editor-plugins/lib/plugins/TableCellSelection/TableCellSelection.ts

```diff
@@ -93,6 +93,13 @@ export default class TableCellSelection implements EditorPlugin {
                         event.rawEvent.preventDefault();
                     }
                     break;
+                case PluginEventType.Input:
+                    if (!this.state.startedSelection) {
+                        this.handleInputEvent(this.editor);
+                    } else {
+                        event.rawEvent.preventDefault();
```

Comment:
Is Input event preventable?

Reply:
It seems we just need to handle keydown event, it will be good enough. I tried to comment out the handling code of input event, and it still works well

Reply:
There is an issue with undo that is mitigated by using the Input method. It only happens if you type one character and do undo, the inputted character and the last written thing will both be undone. This issue is present today and not introduced by this change.

The reason I clear the full cell is that I do not know if select is enough to cover all possible elements in the cell for deletion. And for clearing I want to make sure it is an input and not other keys like caps lock.

Reply:
About the undo issue, can you provide a full end to end repro step? I want to understand more about it. Thanks

Reply:
1. Create a table
2. Fill it with values, remembering which one was the last one
3. Make a cell selection
4. Type one character
5. Undo

Notice that the typed character and the last filled value are both undone. The only one undone should be the typed one.
![tableUndoWrong](https://github.com/microsoft/roosterjs/assets/107568016/9f863984-275c-422d-8a96-c31d2f00022b)

Reply:
Ok, in that case we do need to call addUndoSnapshot() when type with table selection, but only need to do once.

Reply:
And I think it should be good enough to just select everything inside the table cell when keydown. Because:
1. User made a table selection, then type
2. User selected everything in a table cell, then type

Both of them should get the same result. So making a selection to the first select cell means we convert 2 to 1, so that is the right operation.

Reply:
I have been trying both selection types, but each have an issue. editor.select with the Td as parameter has the above issue, while editor.select with the table and coordinates as parameter has an issue of replacing everything in the cell when typing. For example if the cell has multiple lines, only the first one is replaced.

Reply:
I made changes. No I select all child nodes of the cell. I have not found how to solve the above issue with undo.